app/lib/task/backend.dart

```diff
@@ -289,8 +289,14 @@ class TaskBackend {
     // Map from package to updated that has been seen.
     final seen = <String, DateTime>{};
 
+    // The scan process will give up control after 6 hours and the restart
+    // will trigger the 3-days period scan at the beginning of the loop.
+    final stopAfter = clock.fromNow(hours: 6);
```

Comment:
If we want this the behavior of `_scanForPackageUpdates` changes, so documentation comments **must** be updated accordingly.

If you don't want to write a longer documentation comment that explains how this function weirdly stops running every 6 hours. Then perhaps we should structure it differently :rofl: